Source/LibationSearchEngine/LuceneExtensions.cs

```diff
@@ -10,21 +10,47 @@ namespace LibationSearchEngine
 {
     // field names are case specific and, due to StandardAnalyzer, content is case INspecific
     internal static class LuceneExtensions
-    {
-        internal static void AddRaw(this Document document, string name, string value)
-            => document.Add(new Field(name, value, Field.Store.YES, Field.Index.NOT_ANALYZED));
+	{
+		internal static void AddAnalyzed(this Document document, string name, string value)
+		{
+			if (value is not null)
+				document.Add(new Field(name.ToLowerInvariant(), value, Field.Store.YES, Field.Index.ANALYZED));
+		}        
 
-        internal static void AddAnalyzed(this Document document, string name, string value)
-        {
-            if (value is not null)
-                document.Add(new Field(name.ToLowerInvariant(), value, Field.Store.YES, Field.Index.ANALYZED));
-        }
+        internal static void RemoveRule(this Document document, IndexRule rule)
+		{
+			// fields are key value pairs. MULTIPLE FIELDS CAN POTENTIALLY HAVE THE SAME KEY.
+			// ie: must remove old before adding new else will create unwanted duplicates.
+			foreach (var name in rule.FieldNames)
+				document.RemoveFields(name.ToLowerInvariant());
+		}
 
-        internal static void AddNotAnalyzed(this Document document, string name, string value)
-            => document.Add(new Field(name.ToLowerInvariant(), value, Field.Store.YES, Field.Index.NOT_ANALYZED));
+        internal static void AddIndexRule(this Document document, IndexRule rule, LibraryBook libraryBook)
+		{
+			string value = rule.GetValue(libraryBook);
+			if (value is null) return;
 
-        internal static void AddBool(this Document document, string name, bool value)
-            => document.Add(new Field(name.ToLowerInvariant(), value.ToString(), Field.Store.YES, Field.Index.ANALYZED_NO_NORMS));
+			foreach (var name in rule.FieldNames)
+			{
+				// fields are key value pairs and MULTIPLE FIELDS CAN HAVE THE SAME KEY.
+				// splitting authors and narrators and/or tags into multiple fields could be interesting research.
+				// it could allow for more advanced searches, or maybe it could break broad searches.
+
+				// all searching should be lowercase
+				// external callers have the reasonable expectation that product id will be returned CASE SPECIFIC
```

Comment:
This is no longer true

Reply:
In what way?

Reply:
https://github.com/rmcrackan/Libation/issues/618#issuecomment-1584924184

You expect it to be case insensitive, and that's now how the ID field works. Every id is converted to lowercase before indexing.